tests/unit/lms/services/canvas_api/authenticated_test.py

```diff
@@ -0,0 +1,159 @@
+                        {"refresh_token": oauth_token.refresh_token}
+                    ),
+                    schema=Any(),
+                    url_stub=Any(),
+                ),
+                # Repeat call with new access token
+                call(*call_args, headers={"Authorization": "Bearer new_access_token"}),
+            )
+        )
+
+        assert result == "success"
+
+    def test_send_raises_CanvasAPIAccessTokenError_if_it_cannot_refresh(
+        self, authenticated_client, basic_client, oauth_token
+    ):
+        oauth_token.refresh_token = None
+        basic_client.send.side_effect = CanvasAPIAccessTokenError
+
+        with pytest.raises(CanvasAPIAccessTokenError):
+            authenticated_client.send(
+                "METHOD", "/path", sentinel.schema, sentinel.params
+            )
+
+    def test_get_token(self, authenticated_client, basic_client, token_store):
+        token = authenticated_client.get_token("authorization_code")
+
+        assert token == "new_access_token"
+
+        basic_client.send.assert_called_once_with(
+            "POST",
+            "login/oauth2/token",
+            params={
+                "grant_type": "authorization_code",
+                "client_id": sentinel.client_id,
+                "client_secret": sentinel.client_secret,
+                "code": "authorization_code",
+                "redirect_uri": sentinel.redirect_uri,
+                "replace_tokens": True,
+            },
+            schema=CanvasTokenResponseSchema,
+            url_stub="",
+        )
+
+        token_store.save.assert_called_once_with(
+            "new_access_token", "new_refresh_token", "new_expires_in"
+        )
+
+    def test_get_refreshed_token(self, authenticated_client, basic_client, token_store):
+        token = authenticated_client.get_refreshed_token("refresh_token")
+
+        assert token == "new_access_token"
+
+        basic_client.send.assert_called_once_with(
+            "POST",
+            "login/oauth2/token",
+            params={
+                "grant_type": "refresh_token",
+                "client_id": sentinel.client_id,
+                "client_secret": sentinel.client_secret,
+                "refresh_token": "refresh_token",
+            },
+            schema=CanvasTokenResponseSchema,
+            url_stub="",
+        )
+
+        token_store.save.assert_called_once_with(
+            "new_access_token", "new_refresh_token", "new_expires_in"
+        )
+
+    @pytest.fixture
+    def basic_client(self, token_response):
+        basic_api = create_autospec(BasicClient)
+
+        basic_api.send.return_value = token_response
+
+        return basic_api
+
+    @pytest.fixture
+    def token_response(self):
+        return {
+            "access_token": "new_access_token",
+            "refresh_token": "new_refresh_token",
+            "expires_in": "new_expires_in",
+        }
+
+    @pytest.fixture
+    def token_store(self, oauth_token):
```

Comment:
Override the `token_store` provided in `conftest.py` because we want to fake it here.